src/monster.h

```diff
@@ -19,6 +19,19 @@ using mtype_id = string_id<mtype>;
 
 typedef std::map< mfaction_id, std::set< int > > mfactions;
 
+class mon_special_attack : public JsonSerializer {
+public:
+    int cooldown;
+    bool enabled;
+
+    // FIXME optional parameters are stand-in for better syntax in savegame_json.cpp for creating new entries
+    mon_special_attack(int c = 0, bool e = true) : cooldown(c), enabled(e) {};
+
+    using JsonSerializer::serialize;
+    virtual void serialize(JsonOut &jsout) const override;
```

Comment:
I had already implemented deserialization inline elsewhere before I discovered that JsonSerializer existed. I'd prefer to leave it that way, because the loading of this data is more complicated than simple deserialization, due to the possibility of monsters.json having changed, and the possibility of loading data from a previous version of the game.